src/AutoRest.CSharp/Common/Output/Expressions/ValueExpressions/AsExpression.cs

```diff
@@ -0,0 +1,9 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.CSharp.Generation.Types;
+
+namespace AutoRest.CSharp.Common.Output.Expressions.ValueExpressions
+{
+    internal record AsExpression(ValueExpression Inner, CSharpType Type) : ValueExpression;
```

Comment:
Completely understand why we named it `AsExpression` but this could get confusing with the method convention.

I don't know that I have a better suggestion :-(

Reply:
Well I guess the usage will make it clear that this is an expression as opposed to a method that converts something to an expression.